src/index.js

```diff
@@ -0,0 +1,144 @@
+// .. _babel-register: https://github.com/babel/babel/tree/master/packages/babel-register
+//
+// The implementation uses the ``require.extensions`` global in Node. Setting a
+// key in this dictionary adds Node handling for new file extensions. The key is
+// the file suffix string to support, the value is a function to call to load and
+// process the file.
+//
+// But ``require.extenions`` has been deprecated in Node because there is the
+// potential for conflicts when a suffix is used by two modules simultaneously.
+// It also encourages the offloading of compilation to runtime instead of build
+// time.
+//
+// Strictly speaking it should not be used. However:
+//
+// - Only this module uses ``.js.rst`` suffixes.
+// - The Node module containing the ``require.extensions`` implementation is
+//   locked so Node is committed to keeping the current form indefinitely.
+// - Production use cases are encouraged to precompile ``.js.rst`` sources.
+//
+// For ``.js.rst`` loads, we currently only read the file contents and strip the
+// Unicode byte order mark if it is present. This will be extended in following
+// work to extract JavaScript codeblocks from the RestructuredText, preform the
+// compilation and include the output into the running environment.
+//
+// .. code-block:: javascript
+
+    const fs = require('fs')
+    const stripBom = require('strip-bom')
+
+    const loadFile = function(module, filename) {
+      var content = fs.readFileSync(filename, 'utf8')
+      content = stripBom(content)
+
+      console.log(content)
+    }
+
+// Add this ``.js.rst`` file extension handler to ``require.extensions``.
+//
+// .. code-block:: javascript
+
+    if (require.extensions) {
+      require.extensions['.js.rst'] = loadFile
+
+// This is not enough unfortunately since although the ``.js.rst`` extension is
+// registered, the Node module loader isn't able to handle multi-part extensions.
+// For this, we follow CoffeeScript in reimplementing Module::load to support
+// multi-part extension handling.
+//
+//     "This is a horrible thing that should not be required."
+//
+// Start by determining the best extension choice from the filename where best is
+// the file extension with the most parts.
+//
+// There is also the edge case of dotfiles to handle, i.e. a leading dot in the
+// filename is legitimate.
+//
+// Note although there is only one multi-part extension, we cannot simply test
+// for it and then use the existing single-part extension implementation in Node
+// for everything else. This is because other modules may have added multi-part
+// extension implementations too. A specific test approach here would break these
+// other modules.
+//
+// This implementation follows the CoffeeScript directly.
+//
+// .. code-block:: javascript
+
+      const path = require('path')
+      const Module = require('module')
+
+      function findExtension(filename) {
+        var extensions = path.basename(filename).split('.')
```

Comment:
`let`?

Reply:
Definitely! Thanks for the teaching moment, I'm going to remember to do this in future.